wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -498,8 +534,56 @@ impl PhysicalDeviceCapabilities {
         extensions
     }
 
-    fn to_wgpu_limits(&self) -> wgt::Limits {
+    fn to_wgpu_limits(&self, update_after_bind: bool) -> wgt::Limits {
         let limits = &self.properties.limits;
+        let max_sampled_textures = if update_after_bind {
```

Comment:
I suppose we could check for `descriptor_binding_sampled_image_update_after_bind` specifically here instead of the catch-all boolean. Maybe we can gather a bitmask for the adapter, and that would be different bitmask (of UAB types) than we use for the device (since it depends on the requested limits).